modules/iam-assumable-roles-with-saml/main.tf

```diff
@@ -1,12 +1,26 @@
+locals {
+  aws_account_id = var.aws_account_id != "" ? var.aws_account_id : data.aws_caller_identity.current.account_id
+  ids            = compact(distinct(concat(var.provider_ids, [var.provider_id])))
+  identifiers = [
+    for id in local.ids :
```

Comment:
I don't think you need to replace partition and account id in a loop like this. The values in `provider_ids` and/or `provider_id` will be already correct.

Reply:
You're right. I assumed it was needed because of the other implementation. I've removed it.